src/App.tsx

```diff
@@ -1,93 +1,277 @@
+
+const USER_ID = 10548;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<Status>(Status.default);
+  const [disableInput, setDisableInput] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<ErrorTypes | null>(null);
+  const [hasEditTodo, setHasEditTodo] = useState(false);
+  const [todoForUpdate, setTodoForUpdate] = useState<Todo | null>(null);
+  const [idTodoForChange, setIdTodoForChange] = useState<number[]>([]);
+
+  const getVisibleTodos = (statusTodo: Status, todosArr: Todo[]) => {
+    switch (statusTodo) {
+      case Status.active:
+        return todosArr.filter(todo => !todo.completed);
+      case Status.completed:
+        return todosArr.filter(todo => todo.completed);
+      default:
+        return todosArr;
+    }
+  };
+
+  let visibleTodos = getVisibleTodos(status, todos);
+
+  const itemsLeftCount = todos.filter(todo => !todo.completed).length;
+
+  const isAnyTodoCompleted = todos.some(todo => todo.completed === true);
+
+  const haveNotTodos = todos.length === 0;
+
+  async function loadedTodos() {
+    try {
+      const result = await getTodos(USER_ID);
+
+      setTodos(result);
+      setErrorMessage(null);
+    } catch (error) {
+      setErrorMessage(ErrorTypes.ErrorGet);
+    }
+  }
+
+  const handleStatus = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    setStatus(event.currentTarget.dataset.type as Status || Status.default);
+  };
+
+  const handleAddTodo = async (
+    event: React.FormEvent<HTMLFormElement>,
+    input: string,
+    setInput: React.Dispatch<React.SetStateAction<string>>,
+  ) => {
+    event.preventDefault();
+    if (input.trim()) {
+      setDisableInput(true);
+      const todoTempo = {
+        id: 0,
+        userId: USER_ID,
+        title: input,
+        completed: false,
+      };
+
+      todos.splice(todos.length, 1, todoTempo);
+
+      setIdTodoForChange((prev) => [...prev, todoTempo.id]);
+
+      setInput('');
+      try {
+        const createdTodo = await createTodo(USER_ID, {
+          title: input,
+          userId: USER_ID,
+          completed: false,
+        });
+
+        todos.splice(todos.length - 1, 1);
+        await setTodos(prev => [...prev, createdTodo]);
+        setDisableInput(false);
+        setIdTodoForChange([]);
+        setErrorMessage(null);
+      } catch (error) {
+        setErrorMessage(ErrorTypes.ErrorPost);
+        setDisableInput(false);
+      }
+    }
+  };
+
+  const handleUpdateTodo = async (todo: Todo) => {
+    setIdTodoForChange((prev) => [...prev, todo.id]);
+
+    try {
+      await updateTodo(todo.id, {
+        title: todo.title,
+        completed: todo.completed,
+      });
+      setErrorMessage(null);
+      setTodoForUpdate(null);
+      setIdTodoForChange([]);
+    } catch (error) {
+      setErrorMessage(ErrorTypes.ErrorPatch);
+    }
+  };
+
+  const handleRemoveTodo = async (todo: Todo) => {
```

Comment:
you don't use index that you added as argument

Reply:
what index?